src/providers/spatialite/qgsspatialiteproviderconnection.cpp

```diff
@@ -360,7 +360,8 @@ QList<QgsSpatiaLiteProviderConnection::TableProperty> QgsSpatiaLiteProviderConne
         QgsSpatiaLiteProviderConnection::TableProperty property;
         property.setTableName( tableName );
         // Create a layer and get information from it
-        std::unique_ptr< QgsVectorLayer > vl = std::make_unique<QgsVectorLayer>( dsUri.uri(), QString(), QLatin1String( "spatialite" ) );
+        // Use OGR because it's way faster
+        std::unique_ptr< QgsVectorLayer > vl = std::make_unique<QgsVectorLayer>( dsUri.database() + "|layername=" + dsUri.table(), QString(), QLatin1String( "ogr" ), QgsVectorLayer::LayerOptions( false, true ) );
```

Comment:
if the layer of interest is a non-spatial one, you may need to pass the LIST_ALL_TABLES=YES open option, otherwise the OGR SQLite/Spatialite driver won't list it.

```shell
$ printf "id,foo\n1,bar\n" > test.csv

$ ogr2ogr test.sqlite test.csv -dsco spatialite=yes -nlt none

$ ogrinfo test.sqlite
INFO: Open of `test.sqlite'
      using driver `SQLite' successful.

$ ogrinfo test.sqlite -oo LIST_ALL_TABLES=YES
INFO: Open of `test.sqlite'
      using driver `SQLite' successful.
1: ElementaryGeometries (None) [private]
2: KNN (None)
3: SpatialIndex (None) [private]
4: data_licenses (None)
5: geom_cols_ref_sys (None) [private]
6: geometry_columns (None) [private]
7: geometry_columns_auth (None) [private]
8: geometry_columns_field_infos (None) [private]
9: geometry_columns_statistics (None) [private]
10: geometry_columns_time (None) [private]
11: spatial_ref_sys (None) [private]
12: spatial_ref_sys_all (None) [private]
13: spatial_ref_sys_aux (None) [private]
14: spatialite_history (None) [private]
15: sql_statements_log (None) [private]
16: sqlite_sequence (None) [private]
17: test (None)
18: vector_layers (None) [private]
19: vector_layers_auth (None) [private]
20: vector_layers_field_infos (None) [private]
21: vector_layers_statistics (None) [private]
22: views_geometry_columns (None)
23: views_geometry_columns_auth (None) [private]
24: views_geometry_columns_field_infos (None) [private]
25: views_geometry_columns_statistics (None) [private]
26: virts_geometry_columns (None)
27: virts_geometry_columns_auth (None) [private]
28: virts_geometry_columns_field_infos (None) [private]
29: virts_geometry_columns_statistics (None) [private]
```

Reply:
ok, fine then